python/pyspark/mllib/recommendation.py

```diff
@@ -156,6 +162,18 @@ def recommendProducts(self, user, num):
         """
         return list(self.call("recommendProducts", user, num))
 
+    def recommendProductsForUsers(self, num):
+        """
+        Recommends topK products for all users.
+        """
+        return self.call("recommendProductsForUsers", num)
```

Comment:
It won't work because it will return `scala.Tuple2`. We need to add a helper on the Scala side to make it work. Using `DataFrame` should work.